### DISCUSSION TITLE
GrandOrgue in Debian and Ubuntu

### DISCUSSION THREAD

**Discussion by svillemot** [GrandOrgue/grandorgue#1113] [General]
Hi,

As a Debian developer, I have packaged GrandOrgue for Debian, see: https://packages.debian.org/sid/grandorgue

As a consequence, GrandOrgue is already available in Debian unstable and testing, and will be available in future stable Debian releases (starting from Debian “Bookworm” 12, scheduled for 2023).

It will also gradually appear in distributions derived from Debian, such as Ubuntu and Linux Mint. In particular, GrandOrgue 3.6.2 is available in Ubuntu 22.04 LTS that has just been released.

On those systems, GrandOrgue can be installed through the graphical package manager, or through the command line with `apt install grandorgue`. The demonstration organ package is available in a separate `grandorgue-demo` package (that will be automatically pulled in by the main one unless you have disable automatic installation of package recommendations).

Don’t hesitate to report any problem with the packaging, preferably through the Debian bug reporting system (see https://www.debian.org/Bugs/Reporting).

**Comment by William-Purvis** [comment]
I've just downloaded GO on Ubuntu 22.04, with the demo organ, and fired it up. It complains that it can't find Jack,  despite jack
being up and running. Happily handles other things, any suggestions? Any supplementary info required?

**Comment by oleg68** [comment]
@William-Purvis 

Could you provide more info?

1. What version of GO are you trying?
2. What jack server are you using: jackd or pipewire?
3. Did the problem occured only once or it is still reproduced?
4. Where do you see the complain: in the terminal window or in the log window? Could you paste the complain message here?
5. Does GO work and sound normally despite the complain or not?
6. Could you post here the screenshot of Settings->Audio (I'm interested by what sound ports are enabled and what is the exact name of the audio output device, including all prefixes)

**Reply by William-Purvis** [reply]
Hi Oleg, problem resolved - but for info:
GO: GrandOrgue v3.6.2-debian (as distributed with Ubuntu 22.04)
jack:  jackd
reproduce: see below
where: pop-up window
working: see below
screenshot:
![image](https://user-images.githubusercontent.com/32897843/218055743-96427f55-d26b-43f2-8cf4-868094afa69e.png)

OK, image is currently working program, when I first started trying to use GO, the sound port selected was Pa which
was running and working as normal for other things. Pop-up window came up:
![image](https://user-images.githubusercontent.com/32897843/218056558-f1252aca-2603-4667-a903-2ed59aec7a4b.png)
When I changed the setting to just the jack output (as in first and current image) the error didn't pop up, but no sound.
Eventually I figured out I had to manually connect GO to System Output using QJackCtl. 

Once connected, GO works fine. I just have to figure out (or you may be able to tell me) how to achieve this automatically.
Intended use is for this to be our church organ, so it should be a case of switch on, wait until system has booted up and
got GO running, then play. We hope to run the PC headless, so no direct user - PC interaction other than organ console.

Many thanks!

**Reply by William-Purvis** [reply]
Further info:
I thought I'd send the following image snapped from QJackCtl:
![image](https://user-images.githubusercontent.com/32897843/218058795-412a759a-e5f4-4094-a479-183a42a02b0b.png)

Currently system output just feeds the laptop speaker, but on the final version I will have an 8-channel USB DAC, feeding 5 individually powered speakers. Any hints on how to set that up would be welcomed.

**Reply by oleg68** [reply]
This popup was caused by the "Jack" sound port was not enabled in the Settings->Audio.

> I just have to figure out (or you may be able to tell me) how to achieve this automatically.

qjackctl has a built-in session manager that waits for the clients appear and connects them automatically.

Click on the `Patchbay` button on it, describe necessary connections and save them to a file. You can start qjackctl with an active profile specified.

> Intended use is for this to be our church organ, so it should be a case of switch on, wait until system has booted up and
got GO running, then play. We hope to run the PC headless, so no direct user - PC interaction other than organ console.

You can create a starting shell script that suspends pulseaudio, starts jackd, starts qjackctl with an active profile and then starts GrandOrgue.

> We hope to run the PC headless, so no direct user - PC interaction other than organ console.

It is feasible, but it is not very easy. Threre are lots of obstacles to overcome.

> Currently system output just feeds the laptop speaker, but on the final version I will have an 8-channel USB DAC, feeding 5 individually powered speakers. Any hints on how to set that up would be welcomed.

1. If you are planning to use Jack Native only, then I suggest to disable all other sound ports in Settings->Audio.
2. I suggest you use 48000 or even 96000 sample rate instead of 44100, because lots of audio devices aren't support 44100 without resampling and most of GO samplesets are recorded at 48000.
3. I'd upgrade GrandOrgue to 3.9.5 or even to 3.10.0 that is comming soon. You should download it from Github.
